tests/test_message.py

```diff
@@ -30,31 +31,28 @@ async def test_message(self):
                 raw_message=raw_message,
             )
 
-            self.assertEqual(message.text, "Hello world")
-            self.assertEqual(message.user, "user")
-            self.assertEqual(message.target, "default")
-            self.assertEqual(message.raw_event["timestamp"], "01/01/2000 19:23:00")
-            self.assertEqual(message.raw_event["messageId"], "101")
-            with self.assertRaises(TypeError):
+            assert message.text == "Hello world"
+            assert message.user == "user"
+            assert message.target == "default"
+            assert message.raw_event["timestamp"] == "01/01/2000 19:23:00"
+            assert message.raw_event["messageId"] == "101"
+            with pytest.raises(TypeError):
                 await message.respond("Goodbye world")
 
-    def test_depreacted_properties(self):
+@pytest.fixture
+def test_depreacted_properties(message):
         message = Message(text="hello", user="user", room="", connector="")
 
         message.target = "spam"
-        with self.assertWarns(DeprecationWarning):
```

Comment:
I haven't tested this, but this code might check that the deprecation warning was triggered:

```python
import warnings
# ...

def test_deprecated_properties():
    with warnings.catch_warnings(record=True) as w:
        message = Message(text="hello", user="user", room="", connector="")
        message.target = "spam"

        assert len(w) == 1
        assert issubclass(w[-1].category, DeprecationWarning)
```

Here's the [StackOverflow answer](https://stackoverflow.com/a/3892301) for context.

Do you have time t work on this? Otherwise I'm happy to merge it once the fixture issue is fixed and I'll open a new PR to add these tests.

Reply:
Im not sure about this. I think I will just try to resolve the issues that fail the tests.